torch/distributed/pipelining/_IR.py

```diff
@@ -843,207 +862,48 @@ def move_param_to_callee(
         attr_nodes = list(filter(lambda n: n.op == "get_attr", split.graph.nodes))
         for node in attr_nodes:
             # Check whether the parameter is used in only one submodule
-            if len(node.users) == 1:
-                user = next(iter(node.users))
+            if len(node.users) > 1:
+                logger.info(
+                    f"Parameter {node.target} used in multiple stages: {node.users}."  # noqa: G004
+                )
+            for user in node.users:
                 assert user.op == "call_module"
                 # Move parameter into submodule
                 move_param_to_callee(
```

Comment:
does this affect the fqn of the shared parameter?

Reply:
No. This PR targets parameters (single FQN) used by multiple stages once the original model is split.

Reply:
@pianpwk 's PR targets the tied parameter case (aliasing):
https://github.com/pytorch/pytorch/pull/127094